scripts/doindex/parsers/docs.mjs

```diff
@@ -0,0 +1,169 @@
+
+    return Promise.all([...chapters].map(async (chapter) => {
+        const title = $(chapter).find([
+            `> ${TITLE_SELECTOR}`,
+            `> .collapse > .collapse__title > ${TITLE_SELECTOR}`
+        ].join(','));
+
+        return {
+            titleNode: title[0],
+            title: title.text(),
+            content: null
+        };
+    }));
+}
+
+/**
+ * @param {import('cheerio').CheerioAPI} $
+ * @param {string} url
+ * @param {Object.<key, *>} data
+ */
+async function docs($, url, data) {
+    const body = $('body[data-template]');
+    const isArtile = body.attr('data-template') === 'article';
+
+    if (!isArtile) {
+        console.log(`skip: /${url} not support for landing layouts`);
+        return [];
+    }
+
+    const article = $('article.article');
+    const pageUrl = new URL($('link[rel="canonical"]').attr('href'));
+
+    dropUiElements(article);
+    dropNextSections(article);
+    processMedia(article, pageUrl.toString());
+
+    const breadcrumbs = getBreadcrumbs(body);
+
+    const [pageIntro, chaptersTopLevel] = await Promise.all([
+        getIntroduction($, article),
+        getChapters($, article)
+    ]);
+
+    const chapters = await Promise.all(
+        chaptersTopLevel.map(async ({ title, titleNode }) => {
+            const chapterNode = $(titleNode.parentNode);
+
+            const [chapterIntro, chaptersSubLevel] = await Promise.all([
+                getIntroduction($, chapterNode),
+                getChapters($, chapterNode)
+            ]);
+
+            const subChapters = [chapterIntro].concat(...await Promise.all(
+                chaptersSubLevel.map(async subChapter => {
+                    const subTitleNode = subChapter.titleNode;
+                    const subChapterNode = $(subTitleNode.parentNode);
+                    const subContentNode = subChapterNode.is('.collapse__title') ?
+                        findNextElementWith(subChapterNode[0], el =>
+                            $(el).is('.collapse__content, .collapse__body')) :
+                        subTitleNode.nextSibling;
+
+                    return {
+                        titleNode: subTitleNode,
+                        title: title ? [title, subChapter.title] : subChapter.title,
+                        content: await htmlToText($, [subContentNode])
+                    };
+                }))
+            );
+
+            return subChapters;
+        })
+    );
+
+    const subArticles = [pageIntro].concat(...chapters);
+
+    return Promise.all(subArticles.map(async ({ title, titleNode, content }) => {
+        const id = $(titleNode).attr('id');
+        const finalUrl = `/${url}#${id}`;
+        return ({
+            ...DEFAULT_RECORD,
+            ...data,
+
+            objectID: finalUrl,
+            pageType: 'docs',
+            url: new URL(finalUrl, pageUrl),
+            parent: '/' + url,
+
+            headings: breadcrumbs.concat(title).reverse().join(' | '),
```

Comment:
1st word has priority before 2nd in search result order.
Logically heading should be reversed.